src/Storages/HDFS/ReadBufferFromHDFS.cpp

```diff
@@ -63,6 +64,7 @@ struct ReadBufferFromHDFS::ReadBufferFromHDFSImpl : public BufferWithOwnMemory<S
             throw Exception(ErrorCodes::CANNOT_OPEN_FILE,
                 "Unable to open HDFS file: {}. Error: {}",
                 hdfs_uri + hdfs_file_path, std::string(hdfsGetLastError()));
+        file_size = getFileSize();
```

Comment:
getFileSize can be used several times now (some formats use it for example), so let's cache the result file size and return it in getFileSize() it is was already fetched.

Reply:
Yes,  So I use getFileSize to get and save it,  and when before read the file, compare it with the size already fetched, if it is greater than the file size, return directly.

Reply:
I meant, that method getFileSize is called not only here in constructor, but later it can also be called, but then (in getFileSize()) we can return already saved file size instead of calculating from scratch. For this I'd rather make `getFileSizeImpl()` which will go to hdfs and get it, while just getFileSize will return file_size which is already initialized here in constructor